Set4/29/parser/pureDouble.cc

```diff
@@ -0,0 +1,16 @@
+#include "parser.ih"
+
+    // by convert.cc (via number.cc) 
+
+bool Parser::pureDouble(double *dest, string const &str)
+{
+    size_t position;
+    *dest = stod(str, &position);
+    
+    if (position != str.size())
+        return false;
```

Comment:
This needs a comment because it's not obvious what we check here. This is what I wrote:


```cpp
    size_t pos;
    *dest = std::stod(str, &pos);  // may throw - handled by convert()

    if (pos != str.size())
        // pos is the index of the first character in str that was not used
        // by stod to form the double. If it's not equal to str.size(), then
        // we know that not all characters were used.
        return false;
```